source/_posts/2016-02-25-migrating-to-ember-cli-page-object-1-dot-0.markdown

```diff
@@ -0,0 +1,313 @@
+polished the tree structure we already had when defining page objects.
+The `Ceibo` project was born (you can see it over
+[here](http://github.com/san650/ceibo)) which defines a simple way of
+creating complex properties within an object. So for most cases
+properties used only to get a value, will no longer need parentheses
+when accessed.
+
+{% codeblock lang:js %}
+  const page = create({
+    scope: '#my-page',
+
+    title: text('h1'),
+    fillInName: fillable('#name')
+  });
+{% endcodeblock %}
+
+On versions `0.x` the following code was used within tests:
+
+{% codeblock lang:js %}
+  assert.equal(page.title(), 'My page title');
+  page.fillInName('Juan'); // fill #name with 'Juan'
+{% endcodeblock %}
+
+On `v1` this should be changed to:
+
+{% codeblock lang:js %}
+  assert.equal(page.title, 'My page title');
+  page.fillInName('Juan'); // Doesn't change
+{% endcodeblock %}
+
+## Element lookup
+
+On versions `0.x` defining the `scope` attribute on a page object used
+to override how the element was looked up in the DOM. Example:
+
+{% codeblock lang:js %}
+  const page = create({
+    scope: '#my-page',
+
+    title: text('h1'),
+    fillInName: fillable('#name')
+
+    modal: {
+      scope: '#my-modal',
+
+      title: PageObject.text('h3')
+    }
+  });
+{% endcodeblock %}
+
+When running tests on `0.x`:
+
+{% codeblock lang:js %}
+  page.title(); // translates to `find('#my-page h1').text()`
+  page.modal().title() //  transaltes to `find('#my-modal h3').text()`
+{% endcodeblock %}
+
+On `v1` we decided to implement scope inheritance, this means that if a
+component defines a scope and has a child component, the latter will
+inherit its parent scope.
+
+{% codeblock lang:js %}
+  page.title; // translates into find('#my-page h1').text()
+  page.modal.title //  transaltes into find('#my-page #my-modalh3').text()
+{% endcodeblock %}
+
+In some scenarios, this change of behavior will not affect test
+assertions but in some cases, it will. If you want to make sure lookups
+work as on `0.x` you can use the `resetScope` option (you can see more
+options on the documentation [site](http://ember-cli-page-object.js.org/docs/v1.0.x/options)).
+
+Changed definition to keep lookups the same:
+
+{% codeblock lang:js %}
+  const page = create({
+    scope: '#my-page',
+
+    title: text('h1'),
+    fillInName: fillable('#name')
+
+    modal: {
+      scope: '#my-modal',
+      resetScope: true,
+
+      title: text('h3')
+    }
+  });
+{% endcodeblock %}
+
+Another cause of failures with `v1` is that the addon is now asserting
+that only one element is matched when a selector is built.
+
+For example, if the previous page object definition was used with the
+following HTML:
+
+{% codeblock lang:html %}
+<div class="#my-page">
+  <h1>My title</h1>
+  <h1>My other title</h1>
+</div>
+{% endcodeblock %}
+
+The following call will throw an error:
+
+{% codeblock lang:js %}
+  page.title; // Kaboom!
```

Comment:
I would put here the error message as it appears in the console.

`Error: Assertion Failed: "#my-page h1" matched more than one element. If this is not an error use { multiple: true }`